src/js/program/provider-listener.js

```diff
@@ -0,0 +1,147 @@
+import { _isNaN, _isNumber } from 'utils/underscore';
+import { PLAYER_STATE, STATE_IDLE, MEDIA_VOLUME, MEDIA_MUTE,
+    MEDIA_TYPE, PROVIDER_CHANGED, AUDIO_TRACKS, AUDIO_TRACK_CHANGED,
+    MEDIA_RATE_CHANGE, MEDIA_BUFFER, MEDIA_TIME, MEDIA_LEVELS, MEDIA_LEVEL_CHANGED, MEDIA_ERROR,
+    MEDIA_BEFORECOMPLETE, MEDIA_COMPLETE, MEDIA_META, MEDIA_SEEK, MEDIA_SEEKED,
+    NATIVE_FULLSCREEN } from 'events/events';
+
+export default function ProviderEventListener(mediaController) {
+    return function (type, data) {
+        const { provider, mediaModel, model, attached } = mediaController;
+        const event = Object.assign({}, data, {
+            type: type
+        });
+
+        const modelForward = () => {
+            model.trigger(type, data);
+        };
+
+        switch (type) {
+            case 'flashThrottle': {
+                const throttled = (data.state !== 'resume');
+                model.set('flashThrottle', throttled);
+                model.set('flashBlocked', throttled);
+            }
+                break;
+            case 'flashBlocked':
+                model.set('flashBlocked', true);
+                return;
+            case 'flashUnblocked':
+                model.set('flashBlocked', false);
+                return;
+            case MEDIA_VOLUME:
+                model.set(type, data[type]);
+                return;
+            case MEDIA_MUTE:
+                if (!model.get('autostartMuted')) {
+                    // Don't persist mute state with muted autostart
+                    model.set(type, data[type]);
+                }
+                return;
+            case MEDIA_RATE_CHANGE: {
+                const rate = data.playbackRate;
+                // Check if its a generally usable rate.  Shaka changes rate to 0 when pause or buffering.
+                if (rate > 0) {
+                    model.set('playbackRate', rate);
+                }
+            }
+                return;
+            case MEDIA_TYPE:
+                if (mediaModel.get('mediaType') !== data.mediaType) {
```

Comment:
We can use `MEDIA_TYPE` or `type` in lieu of `'mediaType'` for better minification. There are other types above and below for which we could make this change.